src/sentry/api/endpoints/project_details.py

```diff
@@ -223,6 +227,10 @@ def put(self, request, project):
             project.name = result['name']
             changed = True
 
+        if result.get('platform'):
```

Comment:
Talk to Eric and merging this in as it was when dcramer approved then submitting the platform validation as its own PR